src/Tribe/Views/V2/Views/Widgets/Widget_List_View.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * The List Widget View.
+ *
+ * @package Tribe\Events\Views\V2\Views\Widgets
+ * @since TBD
+ */
+
+namespace Tribe\Events\Views\V2\Views\Widgets;
+
+use Tribe\Events\Views\V2\View;
+use Tribe__Context as Context;
+use Tribe__Utils__Array as Arr;
+
+/**
+ * Class List_Widget_View
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2\Views\Widgets
+ */
+class Widget_List_View extends View {
+
+	/**
+	 * Slug for this view
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected $slug = 'widget-list';
+
+	/**
+	 * Visibility for this view.
+	 *
+	 * @since TBD
+	 *
+	 * @var bool
+	 */
+	protected static $publicly_visible = false;
+
+	/**
+	 * Whether the View should display the events bar or not.
+	 *
+	 * @since TBD
+	 *
+	 * @var bool
+	 */
+	protected $display_events_bar = false;
+
+	/**
+	 * Sets up the View repository arguments from the View context or a provided Context object.
+	 *
+	 * @since TBD
+	 *
+	 * @param  Context|null $context A context to use to setup the args, or `null` to use the View Context.
+	 *
+	 * @return array The arguments, ready to be set on the View repository instance.
+	 */
+	protected function setup_repository_args( Context $context = null ) {
+		$context = null !== $context ? $context : $this->context;
+
+		$args = parent::setup_repository_args( $context );
+
+		$context_arr = $context->to_array();
```

Comment:
Do not do this unless you need the full dump of the Context, and you do not need it here.

When you call `Context::to_array` it will *resolve all locations eagerly* to fill the array.
With TEC, PRO and some other plugins active, that's something like 100+ locations each with its own set of lookups, queries and functions.

Reply:
I am removing as this coding was going to be updated in future tickets. 

I get why it should be removed, but then should we then change it in the list view? Or does that need the full dump of context?
https://github.com/moderntribe/the-events-calendar/blob/aa0bc375b47b86b27c5a2d12acbd09e934658c36/src/Tribe/Views/V2/Views/List_View.php#L237